plugins/inputs/t128_lte/t128_lte.go

```diff
@@ -0,0 +1,340 @@
+package t128_lte
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/fs"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+const (
+	T128_CONFIG_FILE = "/var/run/128technology/config.json"
+	LTE_PATH_DIR     = "/var/run/128technology/lte"
+	GLOBAL_INIT_PATH = "/etc/128technology/global.init"
+	LOCAL_INIT_PATH  = "/etc/128technology/local.init"
+)
+
+var SERVICE_STATUS_MAP = map[string]string{
+	"not-registered-searching": "Unable to Register",
+	"no-service":               "No Service Available",
+	"authentication-failed":    "Authentication Failure",
+	"plmn-not-allowed":         "Forbidden PLMN",
+}
+
+type Collection []interface{}
+
+type DefaultConfig struct {
+	Init struct {
+		RouterName string `json:"routerName"`
+		Id         string `json:"id"`
+	} `json:"init"`
+}
+
+type Nodes struct {
+	Router []struct {
+		Name string `json:"name"`
+		Node []struct {
+			Name string `json:"name"`
+		} `json:"node"`
+	} `json:"router"`
+}
+
+type DeviceInterfaceDetails struct {
+	Config struct {
+		Authority struct {
+			Router []struct {
+				Node []struct {
+					DeviceInterface []struct {
+						Name            string `json:"name"`
+						Type            string `json:"type"`
+						TargetInterface string `json:"targetInterface"`
+					} `json:"deviceInterface"`
+				} `json:"node"`
+			} `json:"router"`
+		} `json:"authority"`
+	} `json:"config"`
+}
+
+type LteMetric struct {
+	Name           string
+	SignalStrength string
+	Carrier        string
+	ModemStatus    string
+	LteBand        string
+	Apn            string
+	ServiceMode    string
+	ServiceStatus  string
+	Imsi           string
+	Imei           string
+	Iccid          string
+}
+
+var configData DefaultConfig
+var nodeData Nodes
+var deviceInterfaceDetails DeviceInterfaceDetails
+var found bool
+
+// T128Lte is an input for metrics of a 128T router instance
+type T128Lte struct {
+	RouterName string `toml:"router_name"`
+	NodeName   string `toml:"node_name"`
+}
+
+var sampleConfig = ``
+
+// SampleConfig returns the default configuration of the Input
+func (*T128Lte) SampleConfig() string {
+	return sampleConfig
+}
+
+// Description returns a one-sentence description on the Input
+func (*T128Lte) Description() string {
+	return "Make a 128T LTE query and return the data"
+}
+
+func (plugin *T128Lte) Init() error {
+	err := plugin.checkConfig()
+	if err != nil {
+		return err
+	}
+	plugin.RouterName, err = plugin.getRouterName()
+	if err != nil {
+		return fmt.Errorf("")
+	}
+	plugin.NodeName, err = plugin.getNodeName()
+	if err != nil {
+		return fmt.Errorf("")
+	}
+	return nil
+}
+
+func (plugin *T128Lte) Gather(acc telegraf.Accumulator) error {
+	return nil
+}
+
+func (c Collection) String() string {
+	var result string
+	for _, metrics := range c {
+		result += fmt.Sprintf("%v\n", metrics)
+	}
+	return result
+}
+
+func (plugin *T128Lte) getMetrics() Collection {
```

Comment:
Get metrics and collection can go away. They are just the python/execd equivalent of `Gather`.

Reply:
got it. I thought to create them separately and keepGather clean.